src/test/ui/consts/miri_unleashed/non_const_fn.stderr

```diff
@@ -0,0 +1,15 @@
+warning: skipping const checks
+  --> $DIR/non_const_fn.rs:8:15
+   |
+LL | const C: () = foo();
+   |               ^^^^^
+
+error[E0080]: evaluation of constant expression failed
+  --> $DIR/non_const_fn.rs:11:22
+   |
+LL |     println!("{:?}", C);
+   |                      ^ referenced constant has errors
```

Comment:
maybe we should add a note for this error in case `allow(const_err)` is set, explaining that the root cause may have been hidden

Reply:
cc @rust-lang/wg-diagnostics not sure if we have that much information or if it would mess with queries or stuff, just a thought

Reply:
Yeah that seems like a good idea

Reply:
Opened an issue: https://github.com/rust-lang/rust/issues/65597

Reply:
> explaining that the root cause may have been hidden

@oli-obk I think it'd make sense to store the const failures in such a way that the warning is no longer needed and the E0080 could display both the current info as well as a span and message similar to the warning's.